test/semantics/tensor.py

```diff
@@ -260,6 +260,13 @@ def test_Tensor_adjoint_eval():
     assert tensor1 == tensor2
 
 
+def test_Tensor_dtype_inference():
+    assert Box("F(A)", Dim(1), Dim(1), data=None) == Box("F(A)", Dim(1), Dim(1))
```

Comment:
```suggestion
    assert Box("F(A)", Dim(1), Dim(1), data=None).dtype == None
```

Otherwise this isn't testing anything!